cli/azd/pkg/ext/hooks.go

```diff
@@ -0,0 +1,379 @@
+				implicitHook.Path,
+				implicitHook.Name,
+			)
+			continue
+		}
+
+		allHooks = append(allHooks, implicitHook)
+	}
+
+	return allHooks
+}
+
+func (h *CommandHooks) getExplicitHooks(prefix HookType, validHookNames []string) []*ScriptConfig {
+	matchingScripts := []*ScriptConfig{}
+
+	// Find explicitly configured hooks from azure.yaml
+	for scriptName, scriptConfig := range h.scripts {
+		if scriptConfig == nil {
+			continue
+		}
+
+		index := slices.IndexFunc(validHookNames, func(hookName string) bool {
+			return hookName == scriptName
+		})
+
+		if index > -1 {
+			// If the script config includes an OS specific configuration use that instead
+			if runtime.GOOS == "windows" && scriptConfig.Windows != nil {
+				scriptConfig = scriptConfig.Windows
+			} else if (runtime.GOOS == "linux" || runtime.GOOS == "darwin") && scriptConfig.Linux != nil {
+				scriptConfig = scriptConfig.Linux
+			}
+
+			scriptConfig.Name = scriptName
+			scriptConfig.Path = strings.ReplaceAll(scriptConfig.Path, "/", string(os.PathSeparator))
+			matchingScripts = append(matchingScripts, scriptConfig)
+		}
+	}
+
+	return matchingScripts
+}
+
+func (h *CommandHooks) getImplicitHooks(prefix HookType, validHookNames []string) []*ScriptConfig {
+	matchingScripts := []*ScriptConfig{}
+
+	hooksDir := filepath.Join(h.cwd, ".azure", "hooks")
+	files, err := os.ReadDir(hooksDir)
+	if err != nil {
+		return matchingScripts
+	}
+
+	// Find implicit / convention based hooks in the .azure/hooks directory
+	// Find `predeploy.sh` or similar matching a hook prefix & valid command name
+	for _, file := range files {
+		fileName := file.Name()
+		fileNameWithoutExt := strings.TrimSuffix(fileName, path.Ext(fileName))
+
+		index := slices.IndexFunc(validHookNames, func(hookName string) bool {
+			return !file.IsDir() && hookName == fileNameWithoutExt
+		})
+
+		if index > -1 {
+			scriptType, err := inferScriptTypeFromFilePath(fileName)
+			if err != nil {
+				log.Printf("Found script hook '%s', but type is not supported. %s\n", fileName, err.Error())
+				continue
+			}
+
+			relativePath, err := filepath.Rel(h.cwd, filepath.Join(hooksDir, fileName))
+			if err != nil {
+				// This err should never happen since we are only looking for files inside the specified cwd.
+				log.Printf("Found script hook '%s', but is outside of project folder. Error: %s\n", file.Name(), err.Error())
+				continue
+			}
+
+			scriptConfig := ScriptConfig{
+				Name:     fileNameWithoutExt,
+				Path:     relativePath,
+				Location: ScriptLocationPath,
+				Type:     scriptType,
+			}
+
+			matchingScripts = append(matchingScripts, &scriptConfig)
+		}
+	}
+
+	return matchingScripts
+}
+
+func (h *CommandHooks) execScript(ctx context.Context, scriptConfig *ScriptConfig) error {
+	// Delete any temporary inline scripts after execution
+	defer func() {
+		if scriptConfig.Location == ScriptLocationInline {
+			os.Remove(scriptConfig.Path)
+		}
+	}()
```

Comment:
Simplify and avoid unnecessary `defer` if not inline script. Note that `defer` is function scoped and not lexically scoped (the `if` block has no effect on execution here).

```suggestion
	if scriptConfig.Location == ScriptLocationInline {
	    defer os.Remove(scriptConfig.Path)
	}
```

Reply:
Hmm - will have to test this.